src/docquery/ocr_processor.py

```diff
@@ -0,0 +1,152 @@
+import abc
+import logging
+from typing import List, Optional, Tuple, Any
+
+import numpy as np
+
+
+class NoOCRReaderFound(Exception):
+    def __init__(self, e):
+        self.e = e
+
+    def __str__(self):
+        return f"Could not load OCR Processor: {self.e}"
+
+
+TESSERACT_AVAILABLE = False
+EASYOCR_AVAILABLE = False
+
+try:
+    import pytesseract  # noqa
+
+    pytesseract.get_tesseract_version()
+    TESSERACT_AVAILABLE = True
+except ImportError:
+    pass
+except pytesseract.TesseractNotFoundError as e:
+    logging.warning("Unable to find tesseract: %s." % (e))
+    pass
+
+try:
+    import easyocr  # noqa
+
+    EASYOCR_AVAILABLE = True
+except ImportError:
+    pass
+
+
+class OCRProcessor(metaclass=abc.ABCMeta):
+    def __init__(self, device: Optional[int]):
+        self.device = device
+        self.check_if_available()
+
+    @abc.abstractmethod
+    def apply_ocr(self, image: "Image.Image") -> Tuple[(List[Any], List[List[int]])]:
+        raise NotImplementedError
+
+    @staticmethod
+    @abc.abstractmethod
+    def check_if_available():
+        raise NotImplementedError
+
+    @staticmethod
+    def normalize_box(box, width, height):
```

Comment:
Since this method is not expected to be called from elsewhere, should we prefix with `_`?

Reply:
It is called in the PDF document class on line 109 to process output of No OCR extraction

Reply:
Ahh ok

Reply:
It's more of a function of the pipeline/model, not the OCR processor, how this normalization works. Maybe we should move it into the `Document` class or `document.py` file?

Reply:
Thats kinda hairy not sure about it. There is different logic to prepare for normalizing. It looks ugly.

Reply:
Well, it seems like (1) each OCR implementation shares the same logic and (2) the PDF implementation also shares it. If that's right, then isn't it something "separate" from OCR?

Reply:
Take a look at what I have done. I agree it should be on the documents.  Tested it on Tesseract and Easyocr works for both.